### PR TITLE
fix: fixed support for micromamba

### PR DESCRIPTION
**Pull request by RR-USAF**
Fix for OPAL-952 support for micromamba

### CODE REVIEW

**Comment by NC-USAF** [comment]
Went over file changes and everything looks good.

**Review by NC-USAF — APPROVED** [review]
Read through everything and it all looks good.

File: environment.yaml [hunk 1]
```diff
@@ -2,19 +2,21 @@ name: conda-vendor-dependencies
...
-  - python
+  - python>=3.7
+  - pip
   - click
   - conda-lock>=1.2.1
-  - conda-build
-  - pip
+  - packaging
+  - requests
   - ruamel.yaml
```
Comment (JR-USAF) [inline environment.yaml:11]: What was the reason for adding/removing these dependencies?

**Review by JR-USAF — CHANGES_REQUESTED** [review]

File: environment.yaml (hunk 1 above)
Comment (RR-USAF) [reply environment.yaml:11]: yeah, i went through the code and they should have been there the whole time.

**Review by JR-USAF — CHANGES_REQUESTED** [review]
I tested the following:

Windows:
pytests
micromamba
mamba
conda

Linux:
pytests
micromamba
mamba
conda

pytests passed for both Windows and Linux. Everything vendored successfully and conda create off of each local channel was successful.

File: conda_vendor/conda_vendor.py [hunk 2]
```diff
@@ -574,14 +591,34 @@ def _get_system_virtual_packages(solver: str) -> List[dict]:
...
-        virtual_packages.append(
-            {"name": pkg[0], "version": pkg[1], "build_string": pkg[2]}
-        )
 
+    if not is_micromamba(solver):
+        for pkg in _json.get("virtual_pkgs", []):
+            virtual_packages.append(
+                    {"name": pkg[0], "version": pkg[1], "build_string": pkg[2]}
+                    )
+    else:
```
Comment (JR-USAF) [inline conda_vendor/conda_vendor.py:600]: Is this what was breaking on Isaacs machine?

**Review by JR-USAF — CHANGES_REQUESTED** [review]

File: conda_vendor/conda_vendor.py (hunk 2 above)
Comment (RR-USAF) [reply conda_vendor/conda_vendor.py:600]: no.  on isaac's machine he hadn't correctly setup his zsh, so wasn't getting his PATH set correctly.

**Review by JR-USAF — APPROVED** [review]

**Pull request merged**